src/Control/Monad/Aff.purs

```diff
@@ -65,9 +65,12 @@ module Control.Monad.Aff
   cancelWith aff c = runFn3 _cancelWith nonCanceler aff c
 
   -- | Converts the asynchronous computation into a synchronous one. All values
-  -- | and errors are ignored.
-  launchAff :: forall e a. Aff e a -> Eff e Unit
-  launchAff = runAff (const (pure unit)) (const (pure unit))
+  -- | are ignored, and if the computation produces an error, it is thrown.
+  launchAff :: forall e a. Aff e a -> Eff (err :: EXCEPTION | e) Unit
```

Comment:
The `EXCEPTION` may be thrown asynchronously, whereas the current signature implies that the returned `Eff` will throw the exception -- and therefore, that such exceptions can be caught using `catchException`. The implication is invalid in the general case, only valid for purely synchronous `Aff`. 

Maybe it suffices to amend the comment to, "... and if the computation produces an error, it is thrown, but not necessarily in the current thread of execution.".